### PR TITLE
Adds NPOESS support to global-workflow

### LINKED ISSUE #1225: Development workflow lacks a special NPOESS job
**Description**
The development (rocoto) workflow lacks an NPOESS job, so `JGFS_ATMOS_PGRB2_SPEC_NPOESS` never gets exercised.

**Requirements**
A new npoess job that runs, either behind a new `DO_NPOESS` switch or behind one of the existing switches that makes sense.

**Acceptance Criteria (Definition of Done)**
A new npoess job that creates the same pgrb2 files as operational.

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
# Description

This PR addresses issue #1225. The following was accomplished:

- A new Rocoto job is added; `jobs/rocoto/npoess.sh`;
- `workflow/applications/gfs_cycled.py` is updated to include the `npoess` task;
- `workflow/applications/gfs_forecast_only.py` is updated to include the `npoess` task;
-  A new function is added to `workflow/rocoto/gfs_tasks.py`, `npoess` is added as new GFS (only) task;
- `workflow/rocoto/tasks.py` is updated to include the `npoess` task.

  Resolves #1225 

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

The resultant Rocoto status for a GFS forecast-only configuration is as follows:

```
user@host:$ rocotorun -w x001_gfsv17_issue_1225.xml -d x001_gfsv17_issue_1225.db
user@host:$ rocotostat -w x001_gfsv17_issue_1225.xml -d x001_gfsv17_issue_1225.db

       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
202103231200             gfsstage_ic         druby://hfe02:36670          SUBMITTING                   -         0           0.0
202103231200                 gfsfcst                           -                   -                   -         -             -
202103231200            gfspost_f000                           -                   -                   -         -             -
202103231200            gfspost_f006                           -                   -                   -         -             -
202103231200            gfspost_f012                           -                   -                   -         -             -
202103231200            gfspost_f018                           -                   -                   -         -             -
202103231200            gfspost_f024                           -                   -                   -         -             -
202103231200             gfsmetpg2g1                           -                   -                   -         -             -
202103231200             gfsmetpg2o1                           -                   -                   -         -             -
202103231200             gfsmetppcp1                           -                   -                   -         -             -
202103231200               gfsnpoess                           -                   -                   -         -             -
202103231200                 gfsarch                           -                   -                   -         -             -
202103231200              gfscleanup   

```

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: parm/config/gfs/config.resources [hunk 1]
```diff
@@ -1013,6 +1013,14 @@ elif [[ ${step} = "awips" ]]; then
     export nth_awips=1
     export memory_awips="3GB"
 
+elif [[ ${step} = "npoess" ]]; then
+
+    export wtime_npoess="03:30:00"
```
Comment (aerorahul) [inline parm/config/gfs/config.resources:1018]: 3 hours and 30 minutes?
On WCOSS2 it takes 2 hours.  Not sure what in this job takes that much time (but that is another discussion).

File: workflow/rocoto/gfs_tasks.py [hunk 2]
```diff
@@ -868,6 +868,18 @@ def gempak(self):
 
         return task
 
+    def npoess(self):
+
+        deps = []
+        dep_dict = {'type': 'metatask', 'name': f'{self.cdump}post'}
```
Comment (aerorahul) [inline workflow/rocoto/gfs_tasks.py]: Looking at the ecflow trigger [here](https://github.com/NOAA-EMC/global-workflow/blob/62d8e99e77ffe44784f4c61c3f2296c6d1a72413/ecf/defs/gfs_00.def#L2095), this job depends on `jgfs_atmos_post_anl` and not the metatask for post-processing forecasts.
I think the dependency here is `postanl`

File: workflow/applications/gfs_forecast_only.py [hunk 3]
```diff
@@ -47,6 +47,9 @@ def _get_app_configs(self):
         if self.do_awips:
             configs += ['awips']
 
+        if self.do_npoess:
+            configs += ['npoess']
```
Comment (aerorahul) [inline workflow/applications/gfs_forecast_only.py]: Based on the ecflow trigger, I don't think this job can run in a forecast-only mode.

**Review by aerorahul — CHANGES_REQUESTED** [review]
mostly looks good.  just some corrections on the dependency of this job and clarification on use in `forecast-only` mode.

**Comment by aerorahul** [comment]
Also missing is `npoess` in the `gfs_cycled` task list for `cdump = gfs`

File: workflow/applications/applications.py [hunk 4]
```diff
@@ -62,6 +62,7 @@ def __init__(self, conf: Configuration) -> None:
         self.do_genesis = _base.get('DO_GENESIS', True)
         self.do_genesis_fsu = _base.get('DO_GENESIS_FSU', False)
         self.do_metp = _base.get('DO_METP', False)
+        self.do_npoess = _base.get('DO_NPOESS', False)
```
Comment (aerorahul) [inline workflow/applications/applications.py:65]: Where is `DO_NPOESS` in `config.base`?

**Review by aerorahul — CHANGES_REQUESTED** [review]
There are some missing updates needed for this to work.

File: parm/config/gfs/config.resources (hunk 1 above)
Comment (HenryRWinterbottom) [reply parm/config/gfs/config.resources:1018]: Admittedly I just copied the `awips` resource above. If there is a suggested value I would be happy to change it. Further, I know nothing about these jobs so the other open PRs, accomplishing similar efforts, might also need to be addressed. See #2082 and #2073.

File: workflow/applications/applications.py (hunk 4 above)
Comment (HenryRWinterbottom) [reply workflow/applications/applications.py:65]: Thank you. I forgot to push this update following testing.

File: parm/config/gfs/config.resources (hunk 1 above)
Comment (aerorahul) [reply parm/config/gfs/config.resources:1018]: your guess is just as good as mine.  I was just baselining with the one in ops.  3:30 is fine.

File: parm/config/gfs/config.npoess [hunk 5]
```diff
@@ -0,0 +1,11 @@
+#! /usr/bin/env bash
+
+########## config.gempak ##########
```
Comment (aerorahul) [inline parm/config/gfs/config.npoess]:
```suggestion
########## config.npoess ##########
```

File: workflow/applications/gfs_forecast_only.py (hunk 3 above)
Comment (HenryRWinterbottom) [reply workflow/applications/gfs_forecast_only.py]: Let's wait on Walter and/or Kate's feedback. I don't know anything about this task.

File: parm/config/gfs/config.npoess (hunk 5 above)
Comment (HenryRWinterbottom) [reply parm/config/gfs/config.npoess]: Thank you.

File: workflow/rocoto/gfs_tasks.py (hunk 2 above)
Comment (HenryRWinterbottom) [reply workflow/rocoto/gfs_tasks.py]: Thank you. Another result of cut-and-paste and not knowing what these jobs do (other than satellite related post-processing).

File: workflow/applications/gfs_cycled.py [hunk 6]
```diff
@@ -89,6 +89,9 @@ def _get_app_configs(self):
         if self.do_awips:
             configs += ['awips']
 
+        if self.do_npoess:
+            configs += ['npoess']
+
```
Comment (aerorahul) [inline workflow/applications/gfs_cycled.py:94]: Down in this file, one will need to add `npoess` to `gfs_tasks` list.

File: workflow/applications/gfs_cycled.py [hunk 7]
```diff
@@ -233,6 +236,9 @@ def get_task_names(self):
         if self.do_awips:
             gfs_tasks += ['awips']
 
+        if self.do_npoesss:
```
Comment (aerorahul) [inline workflow/applications/gfs_cycled.py]: One to many essess.

File: workflow/applications/gfs_forecast_only.py (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply workflow/applications/gfs_forecast_only.py]: Rahul is correct. There is no `postanl` job in forecast-only, so no `npoess`.

**Review by aerorahul — APPROVED** [review]
looks good.

File: parm/config/gfs/config.npoess [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#! /usr/bin/env bash
+
+########## config.npoess ##########
+# GFS NPOESS step specific
+
+echo "BEGIN: config.npoess"
+
+# Get task specific resources
+. "${EXPDIR}/config.resources npoess"
```
Comment (WalterKolczynski-NOAA) [inline parm/config/gfs/config.npoess]:
```suggestion
. "${EXPDIR}/config.resources" npoess
```

File: workflow/rocoto/gfs_tasks.py [hunk 9]
```diff
@@ -868,6 +868,18 @@ def gempak(self):
 
         return task
 
+    def npoess(self):
+
+        deps = []
+        dep_dict = {'type': 'metatask', 'name': f'{self.cdump}postanl'}
```
Comment (WalterKolczynski-NOAA) [inline workflow/rocoto/gfs_tasks.py]: `postanl` is not a metatask
```suggestion
        dep_dict = {'type': 'task', 'name': f'{self.cdump}postanl'}
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: parm/config/gfs/config.npoess (hunk 8 above)
Comment (aerorahul) [reply parm/config/gfs/config.npoess]: Good catch @WalterKolczynski-NOAA

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**